src/use-background-queue/use-background-queue.ts

```diff
@@ -0,0 +1,81 @@
+import { Controller } from '@hotwired/stimulus'
+
+class BackgroundQueueController extends Controller {
+  static queue: string[] | BackgroundQueueOptions[] = []
+}
+
+export interface BackgroundQueueOptions {
+  maxWait?: number,
+  name?: string
+}
+
+const defaultOptions = {}
+const idleCallbackIds: number[] = []
+
+const ric = (fn: Function, maxWait?: number) => {
```

Comment:
I think we should define `window.requestIdleCallback` if it's not defined already instead of wrapping it in a function and returning the function.

As it's outlined in the article you mentioned [here](https://developer.chrome.com/blog/using-requestidlecallback/#checking-for-requestidlecallback).